page-changelog.php

```diff
@@ -15,6 +15,108 @@
 </ul>
 
 <a name="latest10.0"></a></li>
+<h3>Version 10.0.4RC2 <small>Nov 22 2017</small><br></h3>
```

Comment:
let's remove RC2, we usually don't post changelogs here for RCs.
the changelog will be for the final version when it's ready.

The RC changelog is available on central already: https://central.owncloud.org/t/owncloud-server-10-0-4-rc1/10394/1